disk.asm

```diff
@@ -301,12 +301,9 @@ INCLUDED
 
 ; Used registers: A, X, Y
 close_all_logical_files:
-    ldx #0
--   txa
-    pha
-    jsr CLOSE
-    pla
-    tax
-    dex
-    bne -
-    rts
+-   ldx $98    ; number of open files
+    beq +       
+    lda $259,x ; logical file number table
+    jsr CLOSE  ; decrements $98 
+    bcc -
```

Comment:
Let's assume if CLOSE returns with carry set, it indicates an error.

Would it still make sense to close the remaining files?

I.e. replace `bcc -` with `jmp close_all_logical_files`?

Reply:
A small note: For C128, the logical file number table has a different location, so this code is not directly portable.
Instead, the kernal routine CLOSE_ALL could be used to close all open files.

Reply:
That's why you make the big bucks!  $$$

Reply:
...wait. C128 CLOSE_ALL does not work like I thought, it works per-device. So a routine like this is still useful.

Reply:
They couldn't fit this into the Kernal.